contracts/storage/StorageAccessible.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./StorageReadable.sol";
+import "./StorageSimulation.sol";
+
+/**
+ * @title Storage Simulation Library
+ * @author Gnosis Developers
+ * @dev Generic base contract that allows callers to access all internal storage.
+ */
+abstract contract StorageAccessible is StorageReadable, StorageSimulation {
+    /**
+     * @dev Simulates a delegete call to a target contract in the context of self.
+     *
+     * Internally reverts execution to avoid side effects (making it static).
+     * Catches revert and returns encoded result as bytes.
+     *
+     * @param targetContract Address of the contract containing the code to execute.
+     * @param calldataPayload Calldata that should be sent to the target contract (encoded method name and arguments).
+     */
+    function simulate(
+        address targetContract,
+        bytes calldata calldataPayload
+    ) public returns (bytes memory response) {
+        // Suppress compiler warnings about not using parameters, while allowing
+        // parameters to keep names for documentation purposes. This does not
+        // generate code.
+        targetContract;
+        calldataPayload;
+
+        assembly {
+            let internalCalldata := mload(0x40)
+            // Store `simulateAndRevert.selector`.
+            mstore(internalCalldata, "\xb4\xfa\xba\x09")
+            // Abuse the fact that both this and the internal methods have the
+            // same signature, and differ only in symbol name (and therefore,
+            // selector) and copy calldata directly. This saves us approximately
+            // 250 bytes of code and 300 gas at runtime over the
+            // `abi.encodeWithSelector` builtin.
+            calldatacopy(
+                add(internalCalldata, 0x04),
+                0x04,
+                sub(calldatasize(), 0x04)
+            )
+
+            // `pop` is required here by the compiler, as top level expressions
+            // can't have return values in inline assembly. `call` typically
+            // returns a 0 or 1 value indicated whether or not it reverted, but
+            // since we know it will always revert, we can safely ignore it.
+            pop(call(
+                gas(),
+                address(),
```

Comment:
In the safe contract I had to change `address` to something else, and if you want to offload this code to some other contract this might be required (e.g. a wrapper contract or similar). I am not sure what an elegant way would be to support this and this might be a completely different contract in general that allows this (so not required in this PR).

Reply:
This is something that would be nice to be implemented as part of the library I had mentioned in the bi-weekly. That was next on the list :smile: